backend/shork/src/main/kotlin/internal/memory/ResolvedAddresses.kt

```diff
@@ -0,0 +1,8 @@
+package software.shonk.interpreter.internal.memory
+
+internal class ResolvedAddresses(
+    val aFieldRead: Int,
+    val aFieldWrite: Int,
+    val bFieldRead: Int,
+    val bFieldWrite: Int,
```

Comment:
Why is there one for each read and write? Why isn't there just one for each field?

Reply:
Because read and write distance can be different. That's the whole reason this rewrite had to happen in the first place xD